File: windows/src/test/java/com/xushuangbo/clipbridge/windows/util/ServiceAddressFormatterTest.java
package com.xushuangbo.clipbridge.windows.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class ServiceAddressFormatterTest {
    @Test
    void normalize_removesTrailingSlash() {
        assertEquals("https://clipbridge.example.com", ServiceAddressFormatter.normalize("https://clipbridge.example.com/"));
    }

    @Test
    void validate_rejectsMissingScheme() {
        assertEquals("服务地址必须以 http:// 或 https:// 开头", ServiceAddressFormatter.validate("clipbridge.example.com"));
    }

    @Test
    void validate_acceptsNormalHttpsAddress() {
        assertTrue(ServiceAddressFormatter.validate("https://clipbridge.example.com:18444").isBlank());
    }
}

File: android/app/src/test/java/com/xushuangbo/clipbridge/core/network/ServiceAddressFormatterTest.kt
package com.xushuangbo.clipbridge.core.network

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class ServiceAddressFormatterTest {
    @Test
    fun normalize_removesWhitespaceAndTrailingSlash() {
        assertEquals(
            "http://127.0.0.1:18080",
            ServiceAddressFormatter.normalize("  http://127.0.0.1:18080/  "),
        )
    }

    @Test
    fun validate_rejectsAddressWithoutScheme() {
        assertEquals(
            "服务地址必须以 http:// 或 https:// 开头",
            ServiceAddressFormatter.validate("127.0.0.1:18080"),
        )
    }

    @Test
    fun validate_acceptsStandardHttpAddress() {
        assertNull(ServiceAddressFormatter.validate("http://10.0.2.2:18080"))
    }
}
